### PR TITLE
8274521: jdk/jfr/event/gc/detailed/TestGCLockerEvent.java fails when other GC is selected

### PR DESCRIPTION
**Pull request by shipilev**
Simple test bug, need to check if G1 is enabled, or it is a default GC before asking `othervm` with `-XX:+UseG1GC`. Other tests in the same directory do exactly that.

Additional testing:
 - [x] Affected test skipped with Shenandoah now
 - [x] Affected test still passes with G1
 - [x] Affected test still passes default GC

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8274521](https://bugs.openjdk.java.net/browse/JDK-8274521): jdk/jfr/event/gc/detailed/TestGCLockerEvent.java fails when other GC is selected

### Reviewers
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5756/head:pull/5756` \
`$ git checkout pull/5756`

Update a local copy of the PR: \
`$ git checkout pull/5756` \
`$ git pull https://git.openjdk.java.net/jdk pull/5756/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5756`

View PR using the GUI difftool: \
`$ git pr show -t 5756`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5756.diff">https://git.openjdk.java.net/jdk/pull/5756.diff</a>

</details>

### CODE REVIEW

File: test/jdk/jdk/jfr/event/gc/detailed/TestGCLockerEvent.java [hunk 1]
```diff
@@ -26,6 +26,7 @@
  * @test TestGCLockerEvent
  * @key jfr
  * @requires vm.hasJFR
+ * @requires vm.gc == "G1" | vm.gc == null
```
Comment (kimbarrett) [inline test/jdk/jdk/jfr/event/gc/detailed/TestGCLockerEvent.java:29]: I think this should instead be `@requires vm.gc.G1`.  The null check is wrong if running with a build that doesn't include G1 and the default GC is being used.

**Review by kimbarrett — CHANGES_REQUESTED** [review]

File: test/jdk/jdk/jfr/event/gc/detailed/TestGCLockerEvent.java (hunk 1 above)
Comment (shipilev) [reply test/jdk/jdk/jfr/event/gc/detailed/TestGCLockerEvent.java:29]: True, it does look suspicious. I matched what other tests in the same folder are doing. Maybe we should fix them all at once in the followup PR?

Comment (kimbarrett) [reply test/jdk/jdk/jfr/event/gc/detailed/TestGCLockerEvent.java:29]: I don't see a reason to change this test from being wrong to being differently wrong.

Comment (shipilev) [reply test/jdk/jdk/jfr/event/gc/detailed/TestGCLockerEvent.java]: Yeah, you are right. Changed, re-tested. See new commit.

**Review by kimbarrett — APPROVED** [review]
Looks good.

**Review by tschatzl — APPROVED** [review]
Lgtm.

**Comment by shipilev** [comment]
Thanks!

/integrate

**Pull request closed without merging**